src/coreclr/jit/gentree.cpp

```diff
@@ -17685,6 +17685,17 @@ CORINFO_CLASS_HANDLE Compiler::gtGetClassHandle(GenTree* tree, bool* pIsExact, b
         }
     }
 
+    if ((objClass != NO_CLASS_HANDLE) && !*pIsExact && IsTargetAbi(CORINFO_NATIVEAOT_ABI))
+    {
+        CORINFO_CLASS_HANDLE exactClass;
+        if ((info.compCompHnd->getExactClasses(objClass, 1, &exactClass) == 1) &&
+            (info.compCompHnd->compareTypesForCast(exactClass, objClass) == TypeCompareState::Must))
```

Comment:
Why is the cast check needed? Shouldn't classes returned by `getExactClasses` already by definition inherit from/implement the class/interface?

Also, could it be a problem if there is only an abstract class implementing the interface and we return with `*pIsExact = true`? We will not see any non-null instance at runtime, but I'm not sure if the JIT handles it downstream.